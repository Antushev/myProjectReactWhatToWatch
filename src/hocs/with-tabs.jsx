import React, {PureComponent} from 'react';

import {FilmDetailTabsName} from './../utils/const.js';

import FilmDetailTabs from './../components/film-detail-tabs/film-detail-tabs.jsx';

const withTabs = (FilmDetailComponent) => {
  class FilmDetailComponentWithTabs extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activeTabDetailPage: FilmDetailTabsName.OVERVIEW
      };
    }

    render() {
      const {activeTabDetailPage} = this.state;

      return <FilmDetailComponent
        {...this.props}
        activeTab={activeTabDetailPage}
        renderTabs={() => {
          return <FilmDetailTabs
            handleTabClick={(activeTab) => {
              this.setState({
                activeTabDetailPage: activeTab
              });
            }}
          />;
        }}
      />;
    }
  }

  return FilmDetailComponentWithTabs;
};

export {withTabs};
