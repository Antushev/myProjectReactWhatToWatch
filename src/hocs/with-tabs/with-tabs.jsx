import React, {PureComponent} from 'react';

import {FilmDetailTabsName} from '../../utils/const.js';

import FilmDetailTabs from '../../components/film-detail-tabs/film-detail-tabs.jsx';

const withTabs = (Component) => {
  class FilmDetailComponentWithTabs extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activeTabDetailPage: FilmDetailTabsName.OVERVIEW
      };
    }

    render() {
      const {activeTabDetailPage} = this.state;

      return <Component
        {...this.props}
        activeTab={activeTabDetailPage}
        renderTabs={() => {
          return <FilmDetailTabs
            onTabClick={(activeTab) => {
              this.setState({
                activeTabDetailPage: activeTab
              });
            }}
          />;
        }}
      />;
    }
  }

  return FilmDetailComponentWithTabs;
};

export {withTabs};
