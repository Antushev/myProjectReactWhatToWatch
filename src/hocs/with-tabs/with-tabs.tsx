import * as React from 'react';
import {Subtract} from 'utility-types';

import {FilmDetailTabsName} from '../../utils/const';

import FilmDetailTabs from '../../components/film-detail-tabs/film-detail-tabs';

interface State {
  activeTabDetailPage: string;
}

interface InjectingProps {
  activeTab: string;
  renderTabs: () => React.ReactNode;
}

const withTabs = (Component) => {
  type P = React.ComponentProps<typeof Component>;
  type T = Subtract<P, InjectingProps>;

  class FilmDetailComponentWithTabs extends React.PureComponent<T, State> {
    constructor(props) {
      super(props);

      this.state = {
        activeTabDetailPage: FilmDetailTabsName.OVERVIEW
      };
    }

    render() {
      const {activeTabDetailPage} = this.state;

      return <Component
        {...this.props}
        activeTab={activeTabDetailPage}
        renderTabs={() => {
          return <FilmDetailTabs
            onTabClick={(activeTab) => {
              this.setState({
                activeTabDetailPage: activeTab
              });
            }}
          />;
        }}
      />;
    }
  }

  return FilmDetailComponentWithTabs;
};

export {withTabs};
