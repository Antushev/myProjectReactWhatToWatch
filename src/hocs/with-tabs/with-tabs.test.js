import React from 'react';
import renderer from 'react-test-renderer';

import {withTabs} from './with-tabs.jsx';

const MockComponent = () => {
  return <div></div>;
};

const MockComponentWrapped = withTabs(MockComponent);

describe(`Tests HOC with-tabs`, () => {
  it(`Test HOC with-tabs`, () => {
    const tree = renderer
      .create(
          <MockComponentWrapped
            activeTab={`OVERVIEW`}
            renderTabs={() => {}}
          />, {
            createNodeMock() {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
