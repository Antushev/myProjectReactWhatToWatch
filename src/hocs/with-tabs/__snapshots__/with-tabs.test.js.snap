// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests HOC with-tabs Test HOC with-tabs 1`] = `<div />`;
